core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4159,8 +4160,10 @@ public long getGlobalCheckpoint() {
             defaultSettings.getScopedSettings());
         IndexMetaData.Builder builder = IndexMetaData.builder(indexSettings.getIndexMetaData());
         builder.settings(Settings.builder().put(indexSettings.getSettings())
-            .put(IndexSettings.INDEX_TRANSLOG_RETENTION_AGE_SETTING.getKey(), randomFrom("-1", "10nanos", "500ms", "7s", "2m", "60m"))
-            .put(IndexSettings.INDEX_TRANSLOG_RETENTION_SIZE_SETTING.getKey(), randomFrom("-1", "5b", "2kb", "1mb", "50gb"))
+            .put(IndexSettings.INDEX_TRANSLOG_RETENTION_AGE_SETTING.getKey(),
+                randomBoolean() ? "-1" : TimeValue.timeValueMillis(randomNonNegativeLong()).getStringRep())
```

Comment:
Hmm..  not sure about this - I think it's very unlikely we'll end up with a small value that will cause rollovers. Wdyt?